src/rpm/package.rs

```diff
@@ -84,49 +84,28 @@ impl RPMPackage {
         Ok(())
     }
 
-    fn read_and_update_digest_256(
-        hasher: &mut impl digest::Digest,
-        mut source: impl std::io::Read,
-    ) {
-        {
-            // avoid loading it into memory all at once
-            // since the content could be multiple 100s of MBs
-            let mut buf = [0u8; 256];
-            while let Ok(n) = source.read(&mut buf[..]) {
-                if n == 0 {
-                    break;
-                }
-                hasher.update(&buf[0..n]);
-            }
-        }
-    }
-
-    // @todo: delete when EL7 goes EOL, PAYLOADDIGEST has been available for years now.
+    // @todo: delete most of this when EL7 goes EOL, the combination of SHA256HEADER and PAYLOADDIGEST
+    // have been available for a while.  EL7 has SHA1HEADER but not PAYLOADHEADER so the only digest which
+    // can verify both is SIGMD5. EL8+ (and similar distros) have both, so SHA1HEADER isn't needed anymore.
     /// Prepare both header and content digests as used by the `SignatureIndex`.
-    pub(crate) fn create_legacy_header_digests<HC, HACC>(
-        header_cursor: HC,
-        header_and_content_cursor: HACC,
-    ) -> Result<Digests, RPMError>
```

Comment:
This was inefficient because it forced copying everything into a new buffer and repeatedly calling `read()`.  But we already have it all in memory.

Reply:
The function was meant to precisely avoid the need to have anything in memory and work on a reasonable chunk size in memory without having to load everything into memory upfront.

Reply:
Until we actually don't keep the `content` in memory, I'd rather we just kept things simple.  By the time we get around to not doing so, we might not need this code at all anymore, that's my point.

Reply:
I'd feel strongly about it, it's the next think I want to tackle, given a find a few hours to work on it